src/Quotations.ts

```diff
@@ -135,15 +142,16 @@ export function extractFromHtml(messageBody: string, options: ExtractFromHtmlOpt
   // try the checkpoint method again on the stripped-down version.
   if (result1.isTooLong) {
     const result2 = extractQuotationUsingCheckpoints(xmlDocument, messageBody, options);
-    if (result2.didFindQuote && !result2.isTooLong)
+    if (result2.didFindQuote && !result2.isTooLong) {
       return result2;
-
+    }
+    isTooLong = result2.isTooLong;
```

Comment:
@quension-gh do you think this approach makes sense?

Reply:
Yeah I think this works 👍 

This way we know when our chosen limits definitively caused it to give up, which is the easiest thing we can act on. Anything more nuanced seems to be about the content itself, which would require a different kind of investigation anyway.